### DISCUSSION TITLE
tutorials/intro-dgs/10-mutations

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#82] [course feedback]
# tutorials/intro-dgs/10-mutations

https://www.apollographql.com/tutorials/intro-dgs/10-mutations

**Comment by krishankumar-arch** [comment]
Hello Apollo Server Team,

Warm greetings! I have successfully completed the Java DGS course, and I must say the content was outstanding. Now, I am eager to dive into Java-GraphQL development with DGS. I have a few questions regarding the comparison between Java-GraphQL DGS and using Node.js, JavaScript, and TypeScript.

1. Can the Apollo federated server stitch together a subgraph generated by Java-GraphQL DGS with JavaScript subgraphs to create a supergraph? Do you have any performance metrics comparing Java-GraphQL DGS with Node.js subgraphs?

Thank you in advance for your guidance!

Best regards, Krishan K Tyagi

**Reply by lizhennessy** [reply]
Hi there @krishtyagi ! I'm so glad you finished the course and enjoyed the experience.

On the Node.js/TypeScript side, you can definitely take advantage of [Apollo Server](https://www.apollographql.com/docs/apollo-server/performance/caching) and all of its features to get the most out of performance. The DGS framework has a lot of additional [documentation around advanced features](https://netflix.github.io/dgs/advanced/operation-caching/) that we don't explore in the course. While I don't have any specific metrics comparing how Node and DGS perform against one another (it's a great idea to put some together!!!), they absolutely _can_ function as _distinct_ subgraphs within the _same_ supergraph. In just a few weeks we'll be shipping a follow-up course that will delve into this situation exactly: we'll take our DGS server and compose it together with a Node.js subgraph into a brand-new supergraph. 

And though it doesn't use Java, the [Odyssey Voyage series](https://www.apollographql.com/tutorials/voyage-part1) provides a great introduction to all of these Federation concepts in the meantime! Do let us know if you have any other questions we can help with, and thanks for learning with us on Odyssey! 🎉
